### PR TITLE
refactor cli

### PR DESCRIPTION
**Pull request by krancour**
Working on #229, which is in-turn an enabled for https://github.com/akuity/kargo/issues/1330, I intend to add at least a couple new flags.

I decided I'd like to get CLI refactoring similar to what was done in Kargo proper out of the way first in order to have a nice, clean starting point.

__Edit:__ This also undoes the awkward bit where the main program looks at argo 0 to decide on running the CLI or the action. The action is now just a hidden command of the CLI.

### CODE REVIEW

File: cmd/action_cmd.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+		Args:   cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return cmdOpts.run(cmd.Context(), cmd.OutOrStdout())
+		},
+	}
+}
+
+// run performs manifest rendering in a GitHub Actions-compatible manner.
+func (o *actionOptions) run(_ context.Context, out io.Writer) error {
+	logger := libLog.LoggerOrDie()
```
Comment (hiddeco) [inline cmd/action_cmd.go:34]: Think this could be moved into `actionOptions`?

Comment (krancour) [reply cmd/action_cmd.go:34]: You mean make `logger` an attribute of `actionOptions`?

I am tentatively ok with that, but where would you propose setting its value?

Comment (hiddeco) [reply cmd/action_cmd.go:34]: Either set it in the constructor, or pass it from the root command. My main concern with this suggestion was to prepare the command as best as we can for potentially testing, by allowing it to be an injectable dependency.

Comment (krancour) [reply cmd/action_cmd.go]: Fixed in 2bdea8b60480c9a951dca3eb7704def5d7f9e465

**Review by hiddeco — APPROVED** [review]
Based on the recent changes made to Kargo, and while taking note that I still need to ramp up my knowledge of this code base, this LGTM.

**Comment by hiddeco** [comment]
(Linter does not agree though, but that should be addressable).

**Comment by krancour** [comment]
> (Linter does not agree though, but that should be addressable).

Whoops. Something snuck in from another branch I was working on.

**Pull request merged**